### ISSUE TITLE
Implement support for multiple Last.fm compatible API endpoints.

### ISSUE THREAD

**Issue by shymega** [RoxasShadow/rustfm#2]
Currently, I am working on [librefm-rs], but then I found this crate, and I think it would be better to rework this crate to support multiple API endpoints that support the Last.fm API. This would be similar to [pylast], another Last.fm API library.

Issue #1 would be good to link this together with.

What do you think? Is this worth pursuing? I'm more than happy to work on this as well, but I am constrained by college work as well, so I'd have to balance this.

[pylast]: https://github.com/pylast/pylast
[librefm-rs]: https://github.com/shymega/librefm-rs

**Comment by RoxasShadow** [comment]
Hi @shymega, sorry for the late reply 😞 

I think that carrying forward #1 is worth in order to provide a proper Lastfm crate to the community. Unfortunately, in this period (till the next 5 months actually) I'm very busy and I don't feel to restart contributing to this repository right now. Feel free to do so in your spare time though!

**Comment by shymega** [comment]
@RoxasShadow Haha, I know the feeling with being busy. I can have a little hack on it, sure 😺 

I've got several assignments to go through first, however. They take higher priority.

Were you thinking of a complete rewrite using best practices for #1, or just a bit of a shuffle around?

**Comment by RoxasShadow** [comment]
The current master branch should be a decent foundation to proceed further.

Currently we have only a [User](https://github.com/RoxasShadow/rustfm/blob/master/src/user/mod.rs) which has [RecentTracks](https://github.com/RoxasShadow/rustfm/blob/master/src/user/recent_tracks.rs).

The first thing to do is being sure that everything still work fine. Then we need to complete `RecentTracks` adding the fields that could be missing and then going ahead with the others resources belonging to `User`.

**Comment by shymega** [comment]
Cool.  IIRC, libre.fm has pretty much the same API, *but* by default it responds in XML. Should prob add a option to `build_url` that instructs rustfm to use JSON instead, if we just want to stick to JSON.

(Also, as an alert- you have API keys in the repo..)

**Comment by RoxasShadow** [comment]
> you have API keys in the repo

iirc they are read only or more probably stolen from the internet

**Comment by shymega** [comment]
Right, I see. Thanks for the heads-up, but can I suggest we move the API key to a env var for testing, so if it IS stolen from the internet, it cannot be exploited?

**Comment by RoxasShadow** [comment]
Absolutely.

**Comment by shymega** [comment]
For the public record, the API key is _now_ stored in a env var.

This was fixed in ddb21c6cf02431921a56f3d4ab8180bd8edc088f
